Projects/rym/rick and morty/src/App.jsx

```diff
@@ -0,0 +1,52 @@
+import SearchInput from "./components/search-input/search-input"
+import CharactersList from "./components/characters-list/characters-list"
+import getCharacters from "./services/get_characters";
+import { useState,useEffect } from "react";
+
+function useCharacters() {
+  const [characters, setCharacters] = useState([])
+
+  const searchCharacters = (value='') => getCharacters().then(data => {
+        const charactersFilter = data.results.filter(character =>
+          character.name.toLowerCase().includes(value.toLowerCase())
+        )
+        setCharacters(charactersFilter)
+      }).catch(error => {
+          console.error('Error al obtener los personajes:', error)
+    })
+
+    return {
+      searchCharacters,
+      characters
+    }
+}
+
+function App() {
+  const [searchValue, setSearchValue] = useState('')
+  const {searchCharacters,characters} = useCharacters()
+  useEffect(() => {
+    searchCharacters(searchValue)
+  },[searchValue])
+
+  return (
+    <>
+      <img src="http://rym-alvarzrz.s3-website-us-west-1.amazonaws.com/images/background.jpeg" alt="" />
+      <div>
+        <SearchInput searchValue={searchValue} setSearchValue={setSearchValue}/>
+        <ul>
+          {characters.map(characters => (
+            <CharactersList 
+              key={characters.id} 
+              img={characters.image} 
```

Comment:
Why are you sending each property as a prop

what do you thing to send the whole object `characters`, and manage it in your component?

Reply:
i not understand the idea... we could watch it together

Reply:
you are sending characters.image, characters.name, characters.species, and if you need to send another property you have to declare another prop like `xxxx = characters.xxxx`

What do you think about, instead of set each property, set the full object like `character=character`